src/Functions/FunctionSQLJSON.h

```diff
@@ -40,7 +42,7 @@ class FunctionSQLJSONHelpers
     class Executor
     {
     public:
-        static ColumnPtr run(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, uint32_t parse_depth)
+        static ColumnPtr run(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, uint32_t parse_depth, const bool & return_type_allow_complex)
```

Comment:
I think here we can also pass `context` instead of flag `return_type_allow_complex` and than pass it to `insertResultToColumn`. It will be better because this flag makes sense only for `JSON_VALUE` function.
Also, in general passing bool flags as constant reference is useless, use just `bool`.